racket/collects/racket/treelist.rkt

```diff
@@ -0,0 +1,1334 @@
+#lang racket/base
+(require (for-syntax racket/base
+                     syntax/for-body)
+         (only-in '#%kernel
+                  vector-copy
+                  vector*-copy vector*-append vector*-set/copy)
+         '#%flfxnum
+         racket/hash-code
+         "private/sort.rkt"
+         (only-in "private/for.rkt" prop:stream))
+
+(#%declare #:unsafe)
+
+;; RRB tree implemention
+;;
+;; Based on
+;;
+;;  Phil Bagwell and Tiark Rompf, "RRB Trees: Efficient Immutable Vectors"
+;;
+;;  Nicolas Stucki, Tiark Rompf, Vlad Ureche, Phil Bagwell, "RRB-Trees: Efficient Immutable Vectors"
```

Comment:
The Bagwell & Rompf paper has `RRB-Trees` (with a hyphen), and the Stucki, Rompf, Ureche, & Bagwell paper is `RRB Vector: A Practical General Purpose Immutable Sequence`.

Reply:
Fixed, thanks!